jykuo-love-shiritori/twp
# [frontend] Coupon Section

Reeeeesolves #13 partially

### Changes


###  (I) Coupon 
- [x] [/admin/manageCoupon](http://localhost:5173/admin/manageCoupons)
  - navigate steps: person icon (nav bar) > admin > Global Coupons (admin buttons) 
- [x] [/admin/manageCoupon/couponID](http://localhost:5173/admin/manageCoupons/0)
  - navigate steps: Admin Coupon > any coupon item
- [x] [/admin/manageCoupon/new](http://localhost:5173/admin/manageCoupons/new)
  - navigate steps: Admin Coupon > new coupon
- [x] [/seller/manageCoupon](http://localhost:5173/user/seller/manageCoupons)
  - navigate steps: Sell (nav bar) >All Coupons
- [x] [/seller/manageCoupon/couponID](http://localhost:5173/user/seller/manageCoupons/0)
  - navigate steps: Seller Coupon > any coupon item
- [x] [/seller/manageCoupon/new](http://localhost:5173/user/seller/manageCoupons/new)  
  - navigate steps: Seller Coupon > new coupon
- [x] [/sellerCoupon](http://localhost:5173/sellerID/coupons)
  - navigate steps: TODO, the sellerID path is not implemented yet.
- [x] [/coupon](http://localhost:5173/coupons)
  - navigate steps: Coupons (nav bar)
 
### Testing Approach (if unconventional)


- install the pkg (there're new ones)
  ```sh
  cd fronend
  npm i
  ```
- start the frontend server
  ```sh
  cd frontend
  npm run dev
  ```
- go to the new page by steps above
- check __edit__ coupon pages
  - these pages include: 
    - [/admin/manageCoupon/couponID](http://localhost:5173/admin/manageCoupons/0)
    - [/admin/manageCoupon/new](http://localhost:5173/admin/manageCoupons/new)
    - [/seller/manageCoupon/couponID](http://localhost:5173/user/seller/manageCoupons/0)
    - [/seller/manageCoupon/new](http://localhost:5173/user/seller/manageCoupons/new) 
- check __info__ coupon modal (msg box)
  - check them by clicking coupon in fallowing pages:
    - [/sellerCoupon](http://localhost:5173/sellerID/coupons)
    - [/coupon](http://localhost:5173/coupons)

### Additional context


- didnt implement functions of some components
  - add new coupon
  - modify coupon
  - some buttons in various places
